chanzuckerberg/idseq-web
# Convention for analytics logging in React

# Description

This establishes a convention for making analytics logging calls in React where user events are handled. It breaks the existing convention in `ANALYTICS_EVENT_NAMES`. 

Design goals:
* convention that can quickly and consistently applied to ~250 onClick, onChange, onClose and onSubmit callsites currently in the codebase
* make frontend and backend events easily distinguishable by name
* make event names findable in the codebase by simple string match
* make names machine parseable (i.e. splittable on delimiter)
* make meaning obvious in the terms of anybody who uses the app (*note* this is one reason why we can't simply automate the name generation completely)
* names are highly likely to be globally unique
* include important user groups automatically
* convention can easily be followed by any future React implementors

I put in a few callsites as examples.

# Test

1. Go to bulk upload form, create form and accession viz and do the following:
2. click around and enter text in inputs and submit
3. see no errors in console
4. see segement events fired with chrome extension
